barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -535,22 +594,14 @@ int main(int argc, char* argv[])
             std::string output_path = get_option(args, "-o", vk_path + "_fields.json");
             vk_as_fields(vk_path, output_path);
         } else if (command == "avm_prove") {
-            std::string avm_bytecode_path = get_option(args, "-b", "./target/avm_bytecode.bin");
-            std::string output_path = get_option(args, "-o", "./proofs/avm_proof");
-            std::vector<uint8_t> call_data_bytes{};
-
-            if (flag_present(args, "-d")) {
-                auto const call_data_path = get_option(args, "-d", "./target/call_data.bin");
-                call_data_bytes = read_file(call_data_path);
-            }
-
-            srs::init_crs_factory("../srs_db/ignition");
-
-            std::vector<fr> const call_data = many_from_buffer<fr>(call_data_bytes);
-            auto const avm_bytecode = read_file(avm_bytecode_path);
-            auto const proof = avm_trace::Execution::run_and_prove(avm_bytecode, call_data);
-            std::vector<uint8_t> const proof_bytes = to_buffer(proof);
-            write_file(output_path, proof_bytes);
+            std::filesystem::path avm_bytecode_path = get_option(args, "-b", "./target/avm_bytecode.bin");
+            std::filesystem::path calldata_path = get_option(args, "-d", "./target/call_data.bin");
+            std::filesystem::path crs_path = get_option(args, "-c", "../srs_db/ignition");
+            std::filesystem::path output_path = get_option(args, "-o", "./proofs/avm_proof");
+            avm_prove(avm_bytecode_path, calldata_path, crs_path, output_path);
+        } else if (command == "avm_verify") {
+            std::filesystem::path proof_path = get_option(args, "-p", "./proofs/avm_proof");
+            return avm_verify(proof_path) ? 0 : 1;
```

Comment:
You could simply return "!avm_verify(proof_path)"

Reply:
clang-tidy complains about that because of the [implicit bool -> int conversion](https://clang.llvm.org/extra/clang-tidy/checks/readability/implicit-bool-conversion.html)